ivy.el

```diff
@@ -1946,10 +1950,11 @@ This is useful for recursive `ivy-read'."
               (t
                (push def coll))))
       (when sort
-        (if (and (functionp collection)
-                 (setq sort-fn (ivy--sort-function collection)))
-            (when (not (eq collection 'read-file-name-internal))
-              (setq coll (sort coll sort-fn)))
+        (if (functionp collection)
+            (when (and (not (eq collection 'read-file-name-internal))
+                       (<= (length coll) ivy-sort-max-size)
+                       (setq sort-fn (ivy--sort-function collection caller)))
```

Comment:
Note that this changes the fall-through logic. Previously, the alternative (`else` branch) of the `if` conditional would have been taken if, for example, `(functionp collection)` was satisfied, but `(ivy--sort-function collection)` was not. Now, this combination causes neither of the conditional `if`'s branches to be taken. Is this intentional?

Reply:
Yes but unless I'm wrong, taking the `else` branch would only have had any effect if `(ivy--sort-function caller)` was satisfied. But in that case, `(ivy--sort-function collection caller)` is now satisfied as well, so the effect should be the same I think.

Reply:
OK, thanks for explaining (the logic here has always confused me).